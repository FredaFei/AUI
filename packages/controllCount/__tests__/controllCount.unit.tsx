import * as renderer from 'react-test-renderer'
import React from 'react'
import {shallow, mount} from 'enzyme'

import ControllCount from '../controllCount'

describe('测试 ControllCount 以下行为', () => {
  it('ControllCount 存在', () => {
    const json = renderer.create(<ControllCount/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('ControllCount count/maxCount/minCount/readOnly, 默认值', () => {
    const wrapper = mount(<ControllCount/>)
    expect(wrapper.props().count).toEqual(1)
    expect(wrapper.props().maxCount).toEqual(10)
    expect(wrapper.props().minCount).toEqual(1)
    expect(wrapper.props().readOnly).toEqual(false)
  })
  it('ControllCount 设置 count/maxCount/minCount/readOnly', () => {
    const wrapper = mount(<ControllCount count={4} maxCount={5} minCount={2} readOnly/>)
    expect(wrapper.props().count).toEqual(4)
    expect(wrapper.props().minCount).toEqual(2)
    expect(wrapper.props().maxCount).toEqual(5)
    expect(wrapper.props().readOnly).toEqual(true)
  })

  it('ControllCount 设置 button click,触发 onChange 事件', () => {
    const onChange = jest.fn()
    const wrapper = mount(<ControllCount count={4} onChange={onChange}/>)
    const button1 = wrapper.find('button').at(0)
    const button2 = wrapper.find('button').at(1)
    button1.simulate('click')
    expect(onChange).toHaveBeenCalled();
    button2.simulate('click')
    expect(onChange).toHaveBeenCalled();
  })
  it('ControllCount 设置 button click,不触发 onChange 事件', () => {
    const onChange = jest.fn()
    const wrapper = mount(<ControllCount count={1} onChange={onChange}/>)
    const button1 = wrapper.find('button').at(0)
    button1.simulate('click')
    expect(onChange).not.toHaveBeenCalled();
  })
  it('ControllCount 设置 button click,不触发 onChange 事件', () => {
    const onChange = jest.fn()
    const wrapper = mount(<ControllCount count={10} onChange={onChange}/>)
    const button2 = wrapper.find('button').at(1)
    expect(onChange).not.toHaveBeenCalled();
    button2.simulate('click')
    expect(onChange).not.toHaveBeenCalled();
  })
  it('ControllCount 不触发 onChange 事件', () => {
    const onChange = jest.fn()
    const wrapper = mount(<ControllCount count={4} onChange={onChange} minCount={1} maxCount={10}/>)
    const input = wrapper.find('input')
    input.simulate('blur', {target: {value: 'new value'}})
    expect(onChange).not.toHaveBeenCalled();
    input.simulate('blur', {target: {value: '0'}})
    expect(onChange).not.toHaveBeenCalled();
    input.simulate('blur', {target: {value: '12'}})
    expect(onChange).not.toHaveBeenCalled();
    input.simulate('blur', {target: {value: '2'}})
    expect(onChange).toHaveBeenCalled();
  })

  it('ControllCount 设置 自定义类名', () => {
    const wrapper = shallow(<ControllCount className="test"/>)
    expect(wrapper.find('.am-controllCount').hasClass('test')).toBe(true)
  })
  it('ControllCount 设置 自定义样式', () => {
    const wrapper = shallow(<ControllCount style={{color: 'red'}}/>)
    expect(wrapper.render()).toMatchSnapshot()
  });
})