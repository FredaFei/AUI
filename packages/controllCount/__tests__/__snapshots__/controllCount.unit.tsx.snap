// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 ControllCount 以下行为 ControllCount 存在 1`] = `
<div
  className="am-controllCount "
>
  <button
    className="am-controllCount-btn am-controllCount-next disabled"
    onClick={[Function]}
  >
    -
  </button>
  <input
    onBlur={[Function]}
    readOnly={false}
    type="text"
  />
  <button
    className="am-controllCount-btn am-controllCount-next"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;

exports[`测试 ControllCount 以下行为 ControllCount 设置 自定义样式 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "am-controllCount ",
      "style": "color:red",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "am-controllCount-btn am-controllCount-next disabled",
        },
        "children": Array [
          Object {
            "data": "-",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "type": "text",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "am-controllCount-btn am-controllCount-next",
            },
            "children": Array [
              Object {
                "data": "+",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "type": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Object {
        "attribs": Object {
          "type": "text",
        },
        "children": Array [],
        "name": "input",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "am-controllCount-btn am-controllCount-next",
          },
          "children": Array [
            Object {
              "data": "+",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "am-controllCount-btn am-controllCount-next disabled",
          },
          "children": Array [
            Object {
              "data": "-",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "type": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "am-controllCount-btn am-controllCount-next",
        },
        "children": Array [
          Object {
            "data": "+",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "type": "text",
          },
          "children": Array [],
          "name": "input",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "am-controllCount-btn am-controllCount-next disabled",
            },
            "children": Array [
              Object {
                "data": "-",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "button",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "type": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
